### DISCUSSION TITLE
Monitoring

### DISCUSSION THREAD

**Discussion by lee212** [radical-cybertools/radical.pilot#2308] [Ideas]
This is an extension of radical.pilot to provide monitoring feature of live resource metrics such as CPU core utilization, and I/O performance on HPC platforms.

There are two options how to implement:
1. deploy monitoring agent on each node to capture system usage data, it can be existing tools like Ganglia or new tool we write in python
2.  attach performance tool to an executable for getting task-specific metrics, for example, `jsrun a.out` becomes `jsrun <perf-reports> a.out`

Exclusion
- Profiling tools such as TAU is not considered to integrate

Tentative metrics to measure:
- Multicore utilization
- I/O throughput
- GPU utilization
- *all metrics are corresponding to tasks*

**Comment by mtitov** [comment]
Closed in favor https://github.com/radical-cybertools/radical.pilot/discussions/3006
